File: app/src/main/java/com/example/make_a_thon/view/activity/CCTVActivity.java
package com.example.make_a_thon.view.activity;

import android.os.Bundle;

import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.example.make_a_thon.R;
import com.example.make_a_thon.base_java.BaseActivityJava;
import com.example.make_a_thon.databinding.ActivityCctvBinding;
import com.example.make_a_thon.viewmodel.CCTVViewModel;

public class CCTVActivity extends BaseActivityJava<ActivityCctvBinding, CCTVViewModel> {

    private WebView mWebView; // 웹뷰 선언
    private WebSettings mWebSettings; //웹뷰세팅

    @Override
    protected int getLayoutId() {

        return R.layout.activity_cctv;
    }

    @Override
    protected Class<CCTVViewModel> getViewModel() {

        return CCTVViewModel.class;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // 웹뷰 시작
        mWebView = binding.webview;

        mWebView.setWebViewClient(new WebViewClient()); // 클릭시 새창 안뜨게
        mWebSettings = mWebView.getSettings(); //세부 세팅 등록
        mWebSettings.setJavaScriptEnabled(true); // 웹페이지 자바스클비트 허용 여부
        mWebSettings.setSupportMultipleWindows(false); // 새창 띄우기 허용 여부
        mWebSettings.setJavaScriptCanOpenWindowsAutomatically(false); // 자바스크립트 새창 띄우기(멀티뷰) 허용 여부
        mWebSettings.setLoadWithOverviewMode(true); // 메타태그 허용 여부
        mWebSettings.setUseWideViewPort(true); // 화면 사이즈 맞추기 허용 여부
        mWebSettings.setSupportZoom(false); // 화면 줌 허용 여부
        mWebSettings.setBuiltInZoomControls(false); // 화면 확대 축소 허용 여부
        mWebSettings.setLayoutAlgorithm(WebSettings.LayoutAlgorithm.SINGLE_COLUMN); // 컨텐츠 사이즈 맞추기
        mWebSettings.setCacheMode(WebSettings.LOAD_NO_CACHE); // 브라우저 캐시 허용 여부
        mWebSettings.setDomStorageEnabled(true); // 로컬저장소 허용 여부

        mWebView.loadUrl("http://192.168.137.210:8090/?action=stream"); // 웹뷰에 표시할 웹사이트 주소, 웹뷰 시작
    }
}

File: app/src/main/java/com/example/make_a_thon/view/activity/CCTVActivity.kt
package com.example.make_a_thon.view.activity

import android.os.Bundle

import android.webkit.WebView
import android.webkit.WebViewClient

import com.example.make_a_thon.BR
import com.example.make_a_thon.R
import com.example.make_a_thon.base.activity.BaseActivity
import com.example.make_a_thon.databinding.ActivityCctvBinding
import com.example.make_a_thon.viewmodel.CCTVViewModel

class CCTVActivity : BaseActivity<ActivityCctvBinding, CCTVViewModel>() {

    override fun getLayoutId(): Int {
        return R.layout.activity_cctv
    }

    override fun getViewModel(): Class<CCTVViewModel> {
        return CCTVViewModel::class.java
    }

    override fun getBindingVariable(): Int {
        return BR.viewmodel
    }

    override fun initObserver() {
        with(viewModel) {

        }
    }

    private lateinit var webView: WebView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding.webview.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
                view?.loadUrl(url)
                return true
            }
        }
        webView.loadUrl("https://www.google.co.in/")
    }
}
